x-pack/plugins/apm/public/components/alerting/ui_components/apm_rule_kql_filter/kuery_bar.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { fromKueryExpression } from '@kbn/es-query';
+import React, { useEffect, useState } from 'react';
+import { DataViewBase } from '@kbn/es-query';
+import { QuerySuggestion } from '@kbn/unified-search-plugin/public';
+import { WithKueryAutocompletion } from './with_kuery_autocompletion';
+import { AutocompleteField } from './autocomplete_field';
+
+type LoadSuggestionsFn = (
+  e: string,
+  p: number,
+  m?: number,
```

Comment:
e, p and m? What is that?

Reply:
Refactored the code to reuse KQL filter related components from `observability` plugin.